src/frontend/Navigation/AppNavigator.tsx

```diff
@@ -33,10 +26,7 @@ export const AppNavigator = () => {
       initialRouteName="Home"
       screenOptions={NavigatorScreenOptions}
     >
-      {createDefaultScreenGroup(formatMessage)}
-      {devExperiments.onboarding && createOnboardingScreenGroup(formatMessage)}
-      {devExperiments.appPasscode &&
-        createAppPasscodeScreenGroup(formatMessage)}
+      {ScreensWithAuth(formatMessage)}
```

Comment:
I'm not sure this navigation approach will work for the passcode? I haven't tested this on my device, but will this not reset the navigation state every time you return to the app? (because the screens are all removed from the stack then re-added).

I think what is needed is to navigate to the passcode screen when the appState changes to background? And to maybe set the passcode screen as the first screen when first mounting the app? I'm not sure though, I can think this through with you if helpful.

Reply:
Yea, i think if we do that, then we can change `authState` into `obscureCodeSet:boolean`, since we won't have to conditionally render anything. And then we would have to keep the `AppStateListener` inside the app container so it has access to navigation. But it should simplify things. I am going to try implementing it now

Reply:
I updated the nav structure. It has the desired outcome of saving nav state, and I was able to get rid of authState.

Also when the user closes the app, they have to veer away from the app. when they veer away from the app, they are navigated to auth screen. Since nav state is saved, when they open the app again they are navigated to the auth screen.